### PR TITLE
[DO-NOT-MERGE] CEL `has` bug proof of concept

### PR DESCRIPTION
**Pull request by alexzielenski**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Shows that `has` behaves differently on native types vs CRDS with identical schemas and that the bug is visible through the current ValidatingAdmissionPolicy

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/apiserver/cel/validatingadmissionpolicy_test.go [hunk 1]
```diff
@@ -712,6 +715,158 @@ func TestMultiplePolicyBindings(t *testing.T) {
...
+
+	// Apply a ConfigMap with `false` for non-pointer omitempty field.
+	// It should be converted into unstructured and `has` returns false
+	// And therefore pass the admission policy
+	_, e = dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "deployments",
+	}).Namespace("default").Apply(context.TODO(), "nginx-deployment", &deploy, opts)
+	require.NoError(t, e)
```
Comment (alexzielenski) [inline test/integration/apiserver/cel/validatingadmissionpolicy_test.go:858]: Deployment with `paused: false` evaluates `has(spec.paused)` in the schema to `false`.

File: test/integration/apiserver/cel/validatingadmissionpolicy_test.go [hunk 2]
```diff
@@ -712,6 +715,158 @@ func TestMultiplePolicyBindings(t *testing.T) {
...
+
+	// Apply a CRD ConfigMap with `false` for non-pointer omitempty field
+	// It should retain knowledge that `false` was specified
+	_, e = dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    "stable.example.com",
+		Version:  "v1",
+		Resource: "deployments",
+	}).Namespace("default").Apply(context.TODO(), "nginx-deployment", &deploy2, opts)
+	require.NoError(t, e)
```
Comment (alexzielenski) [inline test/integration/apiserver/cel/validatingadmissionpolicy_test.go:867]: CRD with `paused: false` evaluates has(spec.paused) in the schema to `true`, even though the schema is the same.

File: test/integration/apiserver/cel/validatingadmissionpolicy_test.go (hunk 1 above)
Comment (liggitt) [reply test/integration/apiserver/cel/validatingadmissionpolicy_test.go:858]: is this an issue with omitempty on zero-value fields in go types?

File: test/integration/apiserver/cel/validatingadmissionpolicy_test.go (hunk 2 above)
Comment (liggitt) [reply test/integration/apiserver/cel/validatingadmissionpolicy_test.go:867]: I think this issue is slightly different than the one reported in https://github.com/kubernetes/kubernetes/issues/120226

I *think* that issue was saying a CRD stanza defining an object with specific `properties` and one defining an object as a map (`additionalProperties`) behaved differently w.r.t. `has` for the same JSON input

File: test/integration/apiserver/cel/validatingadmissionpolicy_test.go (hunk 1 above)
Comment (alexzielenski) [reply test/integration/apiserver/cel/validatingadmissionpolicy_test.go:858]: yep, omit empty non-pointers

**Pull request closed without merging**